mongoengine/queryset/queryset.py

```diff
@@ -543,6 +543,60 @@ def update_one(self, upsert=False, write_concern=None, **update):
         return self.update(
             upsert=upsert, multi=False, write_concern=write_concern, **update)
 
+    def modify(self, upsert=False, full_response=False, remove=False, new=False, **update):
+        """Update and return the updated document.
+
+        Returns either the document before or after modification based on `new`
+        parameter. If no documents match the query and `upsert` is false,
+        returns ``None``. If upserting and `new` is false, returns ``None``.
+
+        If the full_response parameter is ``True``, the return value will be
+        the entire response object from the server, including the 'ok' and
+        'lastErrorObject' fields, rather than just the modified document.
+        This is useful mainly because the 'lastErrorObject' document holds
+        information about the command's execution.
+
+        :param upsert: insert if document doesn't exist (default ``False``)
+        :param full_response: return the entire response object from the
+            server (default ``False``, not available for PyMongo 3+)
+        :param remove: remove rather than updating (default ``False``)
+        :param new: return updated rather than original document
+            (default ``False``)
+        :param update: Django-style update keyword arguments
+
+        .. versionadded:: 0.9
+        """
+
+        if remove and new:
+            raise OperationError("Conflicting parameters: remove and new")
+
+        if not update and not upsert and not remove:
+            raise OperationError(
+                "No update parameters, must either update or remove")
+
+        queryset = self.clone()
+        query = queryset._query
+        update = transform.update(queryset._document, **update)
```

Comment:
https://github.com/MongoEngine/mongoengine/blob/master/mongoengine/queryset/base.py#L559 has a conditional: `if not IS_PYMONGO_3 or not remove:`. Should this happen if `remove` is true for us? Also, we eventually want to support pymongo 3+ (mongodb v3.2 requires it). If so, should we backport it in a way where we won't have to backport it again once we make the rest of the codebase compatible w/ pymongo v3.2?